tools/grabber.js

```diff
@@ -118,21 +122,52 @@ var grabBlock = function (web3, blockHashOrNumber, listening) {
     }
 }
 
+var checkParentBlock = function (web3, blockData, recursively) {
+    if (blockData.number) {
+        var parentBlockNumber = blockData.number - 1;
+        Block.findOne({ number: parentBlockNumber }, function (err, b) {
+            if (err) {
+                console.log("Cannot find block in db:", err);
+                grabBlock(web3, parentBlockNumber, true);
+            } else {
+                if (b) {
+                    if (b && blockData.parentHash != b.hash) {
+                        console.log("BLOCK IS BAD, GRABBING IT", parentBlockNumber);
+                        cleanupBlockAndTransactionsThenGrab(web3, parentBlockNumber);
+                        if (recursively) {
+                            console.log("Checking recursively for:", parentBlockNumber);
+                            Block.findOne({ number: parentBlockNumber - 1 }, function (err, b) {
+                                if (err) {
+                                    console.log("Cannot find the block in the db while checking recursively:", err)
+                                } else {
+                                    if (b) {
+                                        checkParentBlock(web3, blockData, true);
+                                    }
+                                }
+                            });
+                        }
 
+                    }
+                } else {
+                    grabBlock(web3, parentBlockNumber, true);
```

Comment:
Can we either grab this one by `parentHash`, or do this check and fetch up top and then still execute the parent hash check afterwards?